CMakeLists.txt

```diff
@@ -170,34 +175,38 @@ if(PHYSFS_BUILD_STATIC)
         # no dll exports from the static library
         target_compile_definitions(physfs-static PRIVATE "PHYSFS_STATIC")
     endif()
-
+    target_include_directories(physfs-static PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>")
+    target_link_libraries(physfs-static PRIVATE ${OPTIONAL_LIBRARY_LIBS} ${OTHER_LDFLAGS})
     set(PHYSFS_LIB_TARGET physfs-static)
-    set(PHYSFS_INSTALL_TARGETS ${PHYSFS_INSTALL_TARGETS} ";physfs-static")
+    list(APPEND PHYSFS_INSTALL_TARGETS "physfs-static")
 endif()
 
 option(PHYSFS_BUILD_SHARED "Build shared library" TRUE)
 if(PHYSFS_BUILD_SHARED)
     add_library(physfs SHARED ${PHYSFS_SRCS})
+    add_library(PhysFS::PhysFS ALIAS physfs)
     set_target_properties(physfs PROPERTIES MACOSX_RPATH 1)
     set_target_properties(physfs PROPERTIES VERSION ${PHYSFS_VERSION})
     set_target_properties(physfs PROPERTIES SOVERSION ${PHYSFS_SOVERSION})
+    set_target_properties(physfs PROPERTIES EXPORT_NAME PhysFS)
```

Comment:
Should "PhysFS" be lowercase here ("physfs") if this becomes the SONAME of the shared library?

Reply:
This is a CMake thingie, so allow me to explain:
- when you do `add_library(physfs ${PHSYSFS_SOURCES})`, you create a `physfs` library target. You can modify this target by the various `target_*` functions.
- with this target, you can then do `add_library(PhysFS::PhysFS ALIAS physfs)`. Then all users of the library can use `PhysFS::PhysFS`. This target cannot be modified (all `target_*` functions will fail).

This line specifically sets the name of the target when you use [`install(TARGETS EXPORT)`](https://cmake.org/cmake/help/latest/command/install.html#targets) + [`install(EXPORT NAMESPACE)`](https://cmake.org/cmake/help/latest/command/install.html#export) or [`export`](https://cmake.org/cmake/help/latest/command/export.html).

So a user of physfs can do the following interchangeably:
```cmake
add_subdirectory(thirdpary/physfs)
target_link_libraries(mygame PRIVATE PhysFS::PhysFS)
```
or
```cmake
find_package(PhysFS REQUIRED)
target_link_libraries(mygame PRIVATE PhysFS::PhysFS)
```

Before this pr, it was possible for users to modify the `physfs` target in the `add_subdirectory` case, but not in the `find_package` case.